processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -54,15 +56,20 @@ public ExtractionFilter(
   {
     final Indexed<String> allDimVals = selector.getDimensionValues(dimension);
     final List<Filter> filters = Lists.newArrayList();
-    if (allDimVals != null) {
-      for (int i = 0; i < allDimVals.size(); i++) {
+    if (allDimVals != null)
+    {
+      for (int i = 0; i < allDimVals.size(); i++)
+      {
         String dimVal = allDimVals.get(i);
-        if (value.equals(fn.apply(dimVal))) {
+        if (value.equals(fn.apply(dimVal)))
```

Comment:
the contract on `ExtractionFn` is to return null instead of empty string, which can never be true here, so we might be missing a test case for when extraction fn returns null.